forms/session_form.go

```diff
@@ -0,0 +1,56 @@
+package forms
+
+import (
+	"google-scraper/helpers"
+	"google-scraper/models"
+
+	"github.com/beego/beego/v2/adapter/logs"
+	"github.com/beego/beego/v2/core/validation"
+)
+
+type SessionForm struct {
+	Email    string `form:"email" valid:"Email; Required"`
+	Password string `form:"password" valid:"Required;"`
+}
+
+var currentUser *models.User
+
+func (sessionForm *SessionForm) Valid(v *validation.Validation) {
+	errMessage := "Incorrect email or password"
+	logMessage := "Setting error on validation failed: "
+
+	user, err := models.FindUserByEmail(sessionForm.Email)
+	if err != nil {
+		err := v.SetError("Email", errMessage)
+		if err == nil {
+			logs.Error(logMessage, err)
+		}
+	} else {
+		err = helpers.CheckPasswordHash(sessionForm.Password, user.HashedPassword)
+		if err != nil {
+			err := v.SetError("Password", errMessage)
+			if err == nil {
+				logs.Error(logMessage, err)
+			}
+		} else {
+			currentUser = user
+		}
+	}
+}
+
+func (sessionForm *SessionForm) Authenticate() (*models.User, error) {
+	validation := validation.Validation{}
+	success, err := validation.Valid(sessionForm)
+
+	if err != nil {
+		logs.Error("Validation error:", err)
+	}
```

Comment:
WHy having the double `if err != nil` and `if !success` 🤔 Can't we do as nothing is returned in the above conditional block:

```suggestion
if err != nil {
  logs.Error("Validation error:", err)
  
  for _, err := range validation.Errors {
     return nil, err
  }
}
```

Reply:
`success, err := validation.Valid(sessionForm)`, `success` will have true/false based on validation.  Even the validation fails, it does not return errors.  It will return errors for others cases(like we did not pass correct struct)

Reply:
Noted.